ast/ast.go

```diff
@@ -187,10 +187,10 @@ type SubGraph struct {
 
 func NewSubGraph(id, l Attrib) (*SubGraph, error) {
 	g := &SubGraph{}
-	if id == nil {
+	if idTyped, ok := id.(ID); id == nil || (ok && len(idTyped) == 0) {
```

Comment:
If we are going to do the ok check, which I think is a great idea, then we might as well return an error if it is not nil and not idTyped, because otherwise we might end up in the else, where id != nil && !ok and try to use idTyped.